### PR TITLE
feat(loadbalancingexporter): optimize metrics and traces export

### PR DESCRIPTION
**Pull request by mat-rumian**
**Description:** <Describe what has changed.>
PR improves exporting process for `metrics` and `traces`. Previously every metric and trace/span was exported individually. This approach wasn't effective (high cpu usage/hundreds or thousands of tcp connections).

Proposed implementation aggregates telemetry data by exporter and endpoint. I've observed that CPU usage decreased at least 4-5x.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** 
- added tests
- tested manually

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by bryan-aguilar** [comment]
Is there any chance we can get benchmarks around this change?

File: exporter/loadbalancingexporter/helpers.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+)
+
+// mergeTraces concatenates multiple ptrace.Traces into a single ptrace.Traces.
+func mergeTraces(traces ...ptrace.Traces) ptrace.Traces {
+	merged := ptrace.NewTraces()
+	empty := ptrace.Traces{}
+
+	for _, trace := range traces {
+
+		if trace == empty {
```
Comment (bogdandrutu) [inline exporter/loadbalancingexporter/helpers.go:18]: Check num spans. Also you should not use zero-initialized pdata, per comments in the package that these instances are invalid.

File: exporter/loadbalancingexporter/helpers.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+				ils.Scope().CopyTo(newILS.Scope())
+				newILS.SetSchemaUrl(ils.SchemaUrl())
+
+				for k := 0; k < ils.Spans().Len(); k++ {
+					span := ils.Spans().At(k)
+					newSpan := newILS.Spans().AppendEmpty()
+					span.CopyTo(newSpan)
+				}
+			}
+		}
```
Comment (bogdandrutu) [inline exporter/loadbalancingexporter/helpers.go:42]: Is the pdata mutable? If yes, you should use `trace.ResourceSpans().At(i).MoveTo(merged.ResourceSpans().AppendEmpty())` or if not use `trace.ResourceSpans().At(i).CopyTo(merged.ResourceSpans().AppendEmpty())`

File: exporter/loadbalancingexporter/helpers.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+	}
+
+	return merged
+}
+
+// mergeTraces concatenates multiple pmetric.Metrics into a single pmetric.Metrics.
```
Comment (bogdandrutu) [inline exporter/loadbalancingexporter/helpers.go:49]: mergeTraces -> mergeMetrics

File: exporter/loadbalancingexporter/helpers.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+package loadbalancingexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/loadbalancingexporter"
+
+import (
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+// mergeTraces concatenates multiple ptrace.Traces into a single ptrace.Traces.
+func mergeTraces(traces ...ptrace.Traces) ptrace.Traces {
```
Comment (bogdandrutu) [inline exporter/loadbalancingexporter/helpers.go:12]: You only call it with 2 traces in non-test code, better to accept to params.

File: exporter/loadbalancingexporter/helpers.go (hunk 1 above)
Comment (mat-rumian) [reply exporter/loadbalancingexporter/helpers.go:18]: Got it, thank :)

File: exporter/loadbalancingexporter/helpers_test.go [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+	trace1 := ptrace.NewTraces()
+	trace2 := simpleTraces()
+
+	mergedTraces := mergeTraces(trace1, trace2)
+
+	require.Equal(t, expectedTraces, mergedTraces)
+}
+
+func TestMergeTraces(t *testing.T) {
```
Comment (jpkrohling) [inline exporter/loadbalancingexporter/helpers_test.go:36]: Can you also add a benchmark? I totally believe you that this brings better performance, but we should track this.

**Review by jpkrohling — APPROVED** [review]
LGTM, thank you for the change! I think a benchmark would have been useful to show the improvement under different circumstances, but I'm happy with the change as it is already.

File: exporter/loadbalancingexporter/helpers_test.go (hunk 5 above)
Comment (jpkrohling) [reply exporter/loadbalancingexporter/helpers_test.go:36]: If you decide to not do the benchmark, or if you prefer doing it on a follow-up PR, let me know.

**Pull request merged**